admin/views.py

```diff
@@ -962,12 +969,34 @@ def resource_filter(resource_type, status):
     list_display = resource_class.list_display
     if is_custom_template:
         # TODO: hardcoding needs to be removed
-        pending_pagination = model.query.filter(model.is_approved == None).order_by(SaleReceiptModel.id).paginate(
-            page=page, per_page=1, error_out=False
-        )
-        all_pagination = model.query.options(joinedload(SaleReceiptModel.versions)).filter(model.is_approved != None).order_by(desc(SaleReceiptModel.receipt_date)).paginate(
-            page=page, per_page=10, error_out=False
-        )
+        if mandi :
+            pending_pagination = model.query.filter(model.is_approved == None,model.mandi_id == mandi_id).order_by(SaleReceiptModel.id).paginate(
+                page=page, per_page=1, error_out=False
+            )
+            all_pagination = model.query.options(joinedload(SaleReceiptModel.versions)).filter(model.is_approved != None,model.mandi_id == mandi_id).order_by(desc(SaleReceiptModel.receipt_date)).paginate(
+                page=page, per_page=10, error_out=False
+            )
+        elif crop:
+            pending_pagination = model.query.filter(model.is_approved == None,model.crop_id == crop_id).order_by(SaleReceiptModel.id).paginate(
+                page=page, per_page=1, error_out=False
+            )
+            all_pagination = model.query.options(joinedload(SaleReceiptModel.versions)).filter(model.is_approved != None,model.crop_id == crop_id).order_by(desc(SaleReceiptModel.receipt_date)).paginate(
+                page=page, per_page=10, error_out=False
+            )
+        elif crop and mandi :
+            pending_pagination = model.query.filter(model.is_approved == None,model.mandi_id == mandi_id,model.crop_id == crop_id).order_by(SaleReceiptModel.id).paginate(
+                page=page, per_page=1, error_out=False
+            )
+            all_pagination = model.query.options(joinedload(SaleReceiptModel.versions)).filter(model.is_approved != None,model.crop_id == crop_id,model.mandi_id == mandi_id).order_by(desc(SaleReceiptModel.receipt_date)).paginate(
+                page=page, per_page=10, error_out=False
+            )
+        else:
+            pending_pagination = model.query.filter(model.is_approved == None).order_by(SaleReceiptModel.id).paginate(
+                page=page, per_page=1, error_out=False
+            )
+            all_pagination = model.query.options(joinedload(SaleReceiptModel.versions)).filter(model.is_approved != None).order_by(desc(SaleReceiptModel.receipt_date)).paginate(
```

Comment:
I see a lot of code duplication in this change. Can we remove code duplication?

Please see the changes in this PR and see how `.all()` function duplication was removed. We should do the same here.
https://github.com/Gramhal-Foundation/listag/pull/140/files

Reply:
done changes